File: EpicWorkout(Android Studio)/app/src/main/java/com/epicworkout/fernaak/epicworkout/Authentication/Forgot_Password_Dialog.java
package com.epicworkout.fernaak.epicworkout.Authentication;

import android.app.Activity;
import android.app.DialogFragment;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.epicworkout.fernaak.epicworkout.R;
import com.google.firebase.auth.FirebaseAuth;

public class Forgot_Password_Dialog extends DialogFragment {

    public Forgot_Password_Dialog(){}

    private Button btnSend;
    EditText input_email;

    FirebaseAuth mFirebaseAuth;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.auth_dialog_signup, container, false);

        btnSend = view.findViewById(R.id.forgot_password_btn);
        input_email = view.findViewById(R.id.forgot_password_input_email);

        btnSend.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                btnSendClick();
            }
        });
        return view;
    }
    private void btnSendClick() {
        if (!(TextUtils.isEmpty(input_email.getText())))
            mFirebaseAuth.sendPasswordResetEmail(input_email.getText().toString());
    }
}

File: EpicWorkout/EpicWorkout/EpicWorkout.Android/Authentication/Forgot_Password_Dialog.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.Gms.Tasks;
using Firebase.Auth;

namespace EpicWorkout.Droid.Authentication
{
    [Activity(Label = "Forgot_Password_Dialog")]
    public class Forgot_Password_Dialog : DialogFragment, IOnCompleteListener
    {
        public Forgot_Password_Dialog()
        { RetainInstance = true; }

        private Button btnSend;
        EditText input_email;

        FirebaseAuth mAuth;

        public override View OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
        {
            var ignored = base.OnCreateView(inflater, container, savedInstanceState);
            var view = inflater.Inflate(Resource.Layout.dialog_forgot_password, container, false);

            //InitFirebase
            mAuth = FirebaseAuth.GetInstance(Login.app);

            //View
            btnSend = view.FindViewById<Button>(Resource.Id.forgot_password_btn);
            input_email = view.FindViewById<EditText>(Resource.Id.forgot_password_email);

            btnSend.Click += BtnSignup_Click;
            return view;
        }

        private void BtnSignup_Click(object sender, EventArgs e)
        {
            if (input_email.Text == "")
                Toast.MakeText(Context, "Input Email", ToastLength.Short);
            else
                mAuth.SendPasswordResetEmail(input_email.Text)
                .AddOnCompleteListener(this);
        }
        public void OnComplete(Task task)
        {
            if (task.IsSuccessful == false)
            {
                Toast.MakeText(Context, "Reset password failed", ToastLength.Short);
            }
            else
            {
                Toast.MakeText(Context, "Reset password link sent to email : " + input_email.Text, ToastLength.Short);
            }
        }
    }
}